daliuge-engine/dlg/deploy/create_dlg_job.py

```diff
@@ -630,26 +645,46 @@ def main():
                             log_parser = LogParser(log_dir)
                             log_parser.parse(out_csv=opts.csv_output)
                         except Exception as exp:
-                            print(
-                                "Fail to parse {0}: {1}".format(log_dir, exp)
-                            )
+                            print("Fail to parse {0}: {1}".format(log_dir, exp))
         else:
             log_parser = LogParser(opts.log_dir)
             log_parser.parse(out_csv=opts.csv_output)
     elif opts.action == 1:
         if opts.logical_graph and opts.physical_graph:
             parser.error(
-                "Either a logical graph or physical graph filename must be specified"
+                "Either a logical graph XOR physical graph filename must be specified"
             )
-        for path_to_graph_file in (opts.logical_graph, opts.physical_graph):
-            if path_to_graph_file and not os.path.exists(path_to_graph_file):
-                parser.error(
-                    "Cannot locate graph file at '{0}'".format(
-                        path_to_graph_file
+        elif opts.logical_graph:
+            path_to_graph_file = opts.logical_graph
+        elif opts.physical_graph:
+            path_to_graph_file = opts.physical_graph
+        if path_to_graph_file and not os.path.exists(path_to_graph_file):
+            parser.error(f"Cannot locate graph file at '{path_to_graph_file}'")
+        else:
+            graph_name = os.path.basename(path_to_graph_file)
+            with open(path_to_graph_file) as f:
+                if opts.logical_graph:
+                    lg_graph = json.loads(f.read())
+                    pgt = pg_generator.unroll(lg_graph, zerorun=opts.zerorun)
+                    pgt = init_pgt_unroll_repro_data(pgt)
+                    reprodata = pgt.pop()
+                    pgt = pg_generator.partition(
+                        pgt=pgt,
+                        algo="metis",
```

Comment:
Did you want to add `opts.algorithm` here, given the new addition to the parser?

Reply:
Yes